osde2e/managed_node_metadata_operator_tests.go

```diff
@@ -63,29 +63,45 @@ var _ = ginkgo.Describe("managed-node-metadata-operator", ginkgo.Ordered, func()
 			// In osde2e, this is a non-CCS OSD GCP cluster with very limited instance type support
 			instanceType = "custom-4-16384"
 		}
+
 		machinepoolBuilder := clustersmgmtv1.NewMachinePool().ID(machinepoolName).InstanceType(instanceType).Replicas(machinepoolReplicaCount)
 		machinepool, err := machinepoolBuilder.Build()
 		Expect(err).Should(BeNil(), "machinepoolBuilder.Build failed")
-		_, err = ocmClusterClient.MachinePools().Add().Body(machinepool).SendContext(ctx)
-		Expect(err).Should(BeNil(), "failed to create machinepool")
-
-		// delete the pool at the end
-		ginkgo.DeferCleanup(ocmClusterClient.MachinePools().MachinePool(machinepoolName).Delete().SendContext)
-
-		// wait for it to be ready
-		err = wait.For(func() (bool, error) {
-			lblSel := resources.WithLabelSelector(labels.FormatLabels(map[string]string{"hive.openshift.io/machine-pool": machinepoolName}))
-			var machineSetList machinev1beta1.MachineSetList
-			if err = k8s.WithNamespace("openshift-machine-api").List(ctx, &machineSetList, lblSel); err != nil {
-				return false, fmt.Errorf("unable to list machinesets: %w", err)
-			}
-			if len(machineSetList.Items) < 1 {
-				return false, nil
+		// Attempt to create the machinepool 3 times, with a slight wait between.
+		// During testing this locally, (sometimes) the MP Add() didn't error, but no MP was created in OCM at all.
+		MAX_RETRIES := 3
+		for i := 1; i <= MAX_RETRIES; i++ {
+			_, err = ocmClusterClient.MachinePools().Add().Body(machinepool).SendContext(ctx)
+			Expect(err).Should(BeNil(), "failed to create machinepool")
+
+			_, err = ocmClusterClient.MachinePools().MachinePool(machinepoolName).Get().Send()
+
+			if err != nil && i != MAX_RETRIES {
```

Comment:
We will need to error out if we reach max retries and still failed. Let's try something like this in which we only create/get the machinepool on retry, that way we don't wait for it to be ready 3 times. (I didn't test this) 

```go
		// Attempt to create the machinepool 3 times, with a slight wait between.
		retryAttempts := 3
		for i := 1; i <= retryAttempts; i++ {
			_, err = ocmClusterClient.MachinePools().Add().Body(machinepool).SendContext(ctx)
			Expect(err).Should(BeNil(), "failed to create machinepool")

			_, err = ocmClusterClient.MachinePools().MachinePool(machinepoolName).Get().SendContext(ctx)
			if err == nil {
				break
			}
			ginkgo.GinkgoLogr.Info("unable to find machinepool object in OCM.. retrying", "err", err, "attempt", i)
			time.Sleep(10 * time.Second)
		}
		Expect(err).Should(BeNil(), "failed to create machinepool after retrying")

		// delete the pool at the end
		ginkgo.DeferCleanup(ocmClusterClient.MachinePools().MachinePool(machinepoolName).Delete().SendContext)

		// wait for it to be ready
		err = wait.For(func() (bool, error) {
			lblSel := resources.WithLabelSelector(labels.FormatLabels(map[string]string{"hive.openshift.io/machine-pool": machinepoolName}))
			var machineSetList machinev1beta1.MachineSetList
			if err = k8s.WithNamespace("openshift-machine-api").List(ctx, &machineSetList, lblSel); err != nil {
				return false, fmt.Errorf("unable to list machinesets: %w", err)
			}
			if len(machineSetList.Items) < 1 {
				return false, nil
			}
			machineSet := machineSetList.Items[0]
			machineSetName = machineSet.GetName()
			return machineSet.Status.ReadyReplicas == *machineSet.Spec.Replicas, nil
		}, wait.WithTimeout(600*time.Second))

```